.github/workflows/generate-and-deploy-with-delete.yml

```diff
@@ -2,31 +2,60 @@
 name: Generate and Sync Files with Delete
 
 on:
-  workflow_dispatch
+  workflow_dispatch:
+    inputs:
+      environment:
+        type: choice
+        description: The environment of the Registry
+        options:
+          - Development
+          - Production
+      bump-versions-branch:
+        type: string
+        description: The branch with the modules and providers to publish (if not specified will use the workflow's ref)
+        required: false
 
 jobs:
   generate-and-sync:
     runs-on: ubuntu-latest
-
-    environment: Production-with-delete
+    environment:
+      name: ${{ inputs.environment }}
 
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v4
+        with:
+          path: main
+
+      - name: Checkout branch with modules and providers
+        if: inputs.bump-versions-branch != ''
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.bump-versions-branch }}
+          path: providers-and-modules
 
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version-file: './src/go.mod'
+          go-version-file: './main/src/go.mod'
```

Comment:
Was that done on purpose? Doesn't match the `go-version-file` in other GH workflows

Reply:
I had to change it because I changed the path of the checked-out code in [here](https://github.com/opentofu/registry/pull/95/files#diff-801c1b85939a1ea35b8b6e405143602ba9d9a08f04bf5280fa35ed59d2657f5eR33).
Before this change, we ran `git checkout` at root (=default active working directory) once:
root -> the code the WF will run (workflow + scripts) + the modules and providers.

Now the hierarchy is a little different:
root -> `repo` (I renamed `main`) -> the code the WF will run (workflow + scripts)
root -> `providers and modules` -> the branch where the modules and providers were pushed to.

That's why the code is inside the `repo` folder now, and we have to treat it accordingly.

Reply:
OK. Maybe a better approach would be to solely rely on the WF ref branch, and not get a custom branch for fetching the modules and providers.

This way, if you run the dev `bump-versions`, which creates a branch, then afterwards you can run the `generate-and-deploy` WF on the created branch. That branch will also include the PR's updated code

Reply:
True. Originally, I thought it'd be nicer if we had the code branch and the providers & modules branch separated. 
For example, I'm making a big change to both the `bump` script and the `generate and publish` script in my branch.
Let's say I run the `bump` script, have a new branch, and want to change something in the `generate and publish` script before I run it. If the branches are not separated, I'll have to remember to push my change to the newly created branch and my original branch, and it's something I can easily forget.
But I can see why it's too confusing. I'll go with your approach @RLRabinowitz

Reply:
@RLRabinowitz implemented your suggetion